File: 2352-equal-row-and-column-pairs/2352-equal-row-and-column-pairs.cpp
class Solution {
public:
    int equalPairs(vector<vector<int>>& grid) {
        int n = grid.size();
        map<vector<int>, vector<int>> row, col;
        for (int i = 0; i < n; i++) {
            vector<int> v;
            for (int j = 0; j < n; j++) {
                v.push_back(grid[i][j]);
            }
            row[v].push_back(i);
        }
        for (int j = 0; j < n; j++) {
            vector<int> v;
            for (int i = 0; i < n; i++) {
                v.push_back(grid[i][j]);
            }
            col[v].push_back(j);
        }
        int res = 0;
        for (auto it: col) {
            res += it.second.size() * row[it.first].size();
        }
        return res;
    }
};

File: 2352-equal-row-and-column-pairs/2352-equal-row-and-column-pairs.java
class Solution {
    public int equalPairs(int[][] grid) {
        int n = grid.length;
        Map<String, Integer> map = new HashMap<>();
        for (int i = 0; i < n; i++) {
            StringBuilder builder = new StringBuilder();
            for (int j = 0; j < n; j++) {
                builder.append(grid[i][j] + "/");
            }
            String str = builder.toString();
            map.put(str, map.getOrDefault(str, 0) + 1);
        }
        int result = 0;
        for (int j = 0; j < n; j++) {
            StringBuilder builder = new StringBuilder();
            for (int i = 0; i < n; i++) {
                builder.append(grid[i][j] + "/");
            }
            String str = builder.toString();
            result += map.getOrDefault(str, 0);
        }
        return result;
    }
}